### PR TITLE
raidboss: convert 05-shb/trial to TS

### PR DESCRIPTION
**Pull request by quisquous**

### CODE REVIEW

File: ui/raidboss/data/05-shb/trial/ruby_weapon-ex.ts [hunk 1]
```diff
@@ -265,22 +291,25 @@ export default {
...
+      type: 'StartsUsing',
       netRegex: NetRegexes.startsUsing({ source: 'Raven\'s Image', id: '4AFF' }),
       netRegexDe: NetRegexes.startsUsing({ source: 'Naels Trugbild', id: '4AFF' }),
       netRegexFr: NetRegexes.startsUsing({ source: 'Spectre De Nael', id: '4AFF' }),
       netRegexJa: NetRegexes.startsUsing({ source: 'ネールの幻影', id: '4AFF' }),
       netRegexCn: NetRegexes.startsUsing({ source: '奈尔的幻影', id: '4AFF' }),
       netRegexKo: NetRegexes.startsUsing({ source: '넬의 환영', id: '4AFF' }),
       condition: (data, matches) => {
-        if (data.role !== 'healer' || data.role !== 'tank')
+        if (data.role !== 'healer' && data.role !== 'tank')
```
Comment (panicstevenson) [inline ui/raidboss/data/05-shb/trial/ruby_weapon-ex.ts:309]: (🤔)

File: ui/raidboss/data/05-shb/trial/ruby_weapon-ex.ts [hunk 2]
```diff
@@ -375,41 +408,48 @@ export default {
         }
 
         const numAdds = Object.keys(colorToAction).length;
-        let suffix;
-        let actionId;
+        let suffix: string | undefined;
+        let actionId: string | undefined;
```
Comment (panicstevenson) [inline ui/raidboss/data/05-shb/trial/ruby_weapon-ex.ts:412]:
```suggestion
        let suffix: string;
        let actionId: string;
```

File: ui/raidboss/data/05-shb/trial/ruby_weapon-ex.ts [hunk 3]
```diff
@@ -1,20 +1,33 @@
...
+import { UnreachableCode } from '../../../../../resources/not_reached';
 import Outputs from '../../../../../resources/outputs';
 import { Responses } from '../../../../../resources/responses';
 import ZoneId from '../../../../../resources/zone_id';
+import { RaidbossData } from '../../../../../types/data';
+import { TriggerSet } from '../../../../../types/trigger';
+
+export interface Data extends RaidbossData {
+  playerMap?: { [name: string]: boolean };
+  seenFlight?: boolean;
```
Comment (panicstevenson) [inline ui/raidboss/data/05-shb/trial/ruby_weapon-ex.ts:12]:
```suggestion
export interface Data extends RaidbossData {
  seenFlight?: boolean;
```

**Review by panicstevenson — APPROVED** [review]

File: ui/raidboss/data/05-shb/trial/ruby_weapon-ex.ts (hunk 1 above)
Comment (quisquous) [reply ui/raidboss/data/05-shb/trial/ruby_weapon-ex.ts:309]: Thanks TypeScript for catching "this condition is always false".

File: ui/raidboss/data/05-shb/trial/ruby_weapon-ex.ts (hunk 2 above)
Comment (quisquous) [reply ui/raidboss/data/05-shb/trial/ruby_weapon-ex.ts]: Mmm, whoops, `actionId` can be undefined here (and then I check it below).

Comment (panicstevenson) [reply ui/raidboss/data/05-shb/trial/ruby_weapon-ex.ts]: Isn't `let variableName: string;` initialized as undefined?

Comment (quisquous) [reply ui/raidboss/data/05-shb/trial/ruby_weapon-ex.ts]: Yes, but you can't assign undefined to it.

**Pull request merged**